packages/svelte/src/compiler/phases/1-parse/state/tag.js

```diff
@@ -263,38 +265,61 @@ function open(parser) {
 		return;
 	}
 
-	if (parser.eat('snippet')) {
-		parser.require_whitespace();
+	if (parser.match('snippet')) {
+		const snippet_declaration_end = find_matching_bracket(parser.template, parser.index, '{');
 
-		const name_start = parser.index;
-		const name = parser.read_identifier();
-		const name_end = parser.index;
+		if (!snippet_declaration_end) {
+			error(start, 'TODO', 'Expected a closing curly bracket');
```

Comment:
should this just be `unexpected-eof`?

```suggestion
			error(parser.template.length, 'unexpected-eof');
```

Reply:
Yeah, you're right